src/Factory/Product/ProductVariantViewFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusLagersystemPlugin\Factory\Product;
+
+use Setono\SyliusLagersystemPlugin\View\Product\ProductVariantView;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class ProductVariantViewFactory implements ProductVariantViewFactoryInterface
+{
+    /** @var string */
+    private $productVariantViewClass;
+
+    public function __construct(string $productVariantViewClass)
+    {
+        $this->productVariantViewClass = $productVariantViewClass;
+    }
+
+    public function create(ProductVariantInterface $variant, string $localeCode): ProductVariantView
+    {
+        $translation = $variant->getTranslation($localeCode);
+
+        /** @var ProductVariantView $variantView */
+        $variantView = new $this->productVariantViewClass();
+        $variantView->id = $variant->getId();
+        $variantView->code = $variant->getCode();
+        $variantView->name = $translation->getName();
+        $variantView->weight = (int) ceil(1000 * $variant->getShippingWeight());
+        $variantView->onHand = $variant->getOnHand();
```

Comment:
Should there be some kind of check for `isTracked()` here? I mean if an ecommerce store has set the tracked to false it either means that it's a digital product or they have lots in stock.

So a suggestion could be: if tracked then set on hand to onHand value else set on hand to 1

Reply:
Ok, thanks for the idea.